docs/source/examples/Layout and Styling.ipynb

```diff
@@ -0,0 +1,1892 @@
+    "- `flex`\n",
+    "- `align_self`\n",
+    "- `align_content`\n",
+    "- `justify_content`\n",
+    "\n",
+    "\n",
+    "### Shorthand css properties\n",
+    "\n",
+    "You may have noticed that certain css properties such as `margin-[top/right/bottom/left]` seem to be missing. The same holds for `padding-[top/right/bottom/left]` etc.\n",
+    "\n",
+    "In fact, you can atomically specify `[top/right/bottom/left]` margins via the `margin` attribute alone by passing the string\n",
+    "\n",
+    "```\n",
+    "margin: 100px 150px 100px 80px;\n",
+    "```\n",
+    "\n",
+    "for a respectively `top`, `right`, `bottom` and `left` margins of  `100`, `150`, `100` and `80` pixels.\n",
+    "\n",
+    "Similarly, the `flex` attribute can hold values for `flex-grow`, `flex-shrink` and `flex-basis`. The `border` attribute is a shorthand property for `border-width`, `border-style (required)`, and `border-color`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Simple examples"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following example shows how to resize a `Button` so that its views have a height of `80px` and a width of `50%` of the available space:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from ipywidgets import *"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget": "0407987e92794f8398d5952851d407a9"
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "s = Button(description='(50% width, 80px height) button')\n",
+    "s.layout.width = '50%'\n",
+    "s.layout.height = '80px'\n",
+    "s"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `layout` property can be shared amongst multiple widgets and assigned directly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget": "68a44717396e4795a65db0246c0a196c"
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "i = Button(description='Another button with the same layout', layout=s.layout)\n",
+    "i"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### More Styling (colors, inner-widget details)\n",
+    "\n",
+    "The `layout` attribute only exposes layout-related css properties for the top-level DOM element of widgets. Individual widgets may expose more styling-related properties, or none. For example, in the case of the `Button` widget, we added a `button_style` attribute that may take 5 different values:\n",
```

Comment:
For example, **the `Button` widget has a** `button_style` attribute that may take 5 different values: